Simple Incremental/Assets/Scripts/Movement.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+public class Movement : MonoBehaviour
+{
+
+    public event MovementFinishedHandler OnMovementFinished;
+    public delegate void MovementFinishedHandler();
+
+    public float moveSpeed;
+    
+    [SerializeField] 
+    private Transform target;
+
+    [SerializeField] 
+    private float terminationDistance = 0.01f;
+
+    private bool shouldMove = true;
+    private bool movementFinished = false;
+   
+    public void StartMovement()
+    {
+        movementFinished = false;
+        shouldMove = true;
+    }
+    
+    public void StopMovement()
+    {
+        shouldMove = false;
+    }
+
+    public void Update()
+    {
+        if (!movementFinished && shouldMove)
+        {
+            if (Vector3.Distance(target.position, transform.position) > terminationDistance)

```

Comment:
Update() and FixedUpdate() methods are usually the place where I prefer to optimize out of the gate because they are called so frequently. Normally I would follow the rule of "don't optimize until performance lags", but here it is pretty easy to optimize, so I'll suggest it.

Distance calculations are actually quite intensive with Vectors since they require square root calculation. Consider adapting to use sqrMagnitude and caching a square of your termination distance in start for something like this:

((target.position - transform.position).sqrMagnitude > squaredTerminationDistance)

Reply:
sqrMagnitude runs much faster than Vector3.Distance because it does not take the CPU-costly step of square-rooting the difference between points.

Reply:
Fixed